### DISCUSSION TITLE
Font size of the pandas table

### DISCUSSION THREAD

**Discussion by itacdonev** [executablebooks/meta#255] [Q&A] [answered]
Hello,

Is there a way to define the font size of the pandas table?

I have used the following 
```
df.style.set_table_attributes('style="font-size: 10px"')
```
which works great within the Jupyter Lab but once the jupyterbook is build the table is quite big (wrt to the markdown text font) and there is no change with respect to the style inside the notebook.
I also added some colors and highlights via the `style` and these are shown just OK in the jupyterbook. 
Thanks!

**Comment by choldgraf** [comment, marked as answer]
I wonder if @jorisvandenbossche could advise on how Pandas manages the CSS that is output to style tables. Is there any way that we could somehow keep that CSS when Jupyter Book executes and builds the page?

**Reply by jorisvandenbossche** [reply]
Normally I would expect this to work. We have some examples in the pandas docs that change font size, eg https://pandas.pydata.org/docs/dev/user_guide/style.html#Using-Table-Styles (although this is using nbsphinx to include it in the sphinx docs, and not through myst, so that might be a difference). 

Can you maybe check if it works for the linked example with jupyterbook? 
It might be the exact way how to add the style that matters (eg the example uses `set_table_styles` and not `set_table_attributes`)

**Reply by choldgraf** [reply]
hmm actually I just tried it and it does seem to work. See a PR to add an example here: https://github.com/executablebooks/jupyter-book/pull/1220

docs here: https://deploy-preview-1220--jupyter-book.netlify.app/content/code-outputs.html

@itacdonev could it just be that the default font size of JB is different, so the difference isn't showing up?

**Reply by itacdonev** [reply]
I ran the complete build without the cache and now the style appears correctly.
Thanks for the help!

**Reply by choldgraf** [reply]
hooray! 🎉

**Reply by AdamRJensen** [reply]
I just started using the Jupyter book theme with Sphinx, which is awesome, but all my tables look way out of proportion. If we agree that this is a common problem, wouldn't it be better to fix this in the Jupyter book theme?

Alternatively, is there a way to set the font size as a global setting, so that you don't have to clutter every line that outputs a table with ".style.set_table_attributes('style="font-size: 10px"')"?

**Reply by choldgraf** [reply]
you can see the default table style here: https://sphinx-book-theme.readthedocs.io/en/latest/notebooks.html#removing-content-before-publishing does that not look correct to you?
